### PR TITLE
Document AddressChecksumCore class

### PR DESCRIPTION
**Pull request by firstred**
| Questions | Answers |
| --- | --- |
| Branch? | `develop` |
| Description? | Upgrading the PHPDoc comments of this class, using the latest PS standard. Also getting the Franglais out of the comments, getting rid of things my Symfony code sniffer nags about, etc. It might have some additional `todo` tags, depending on whether the code was comprehensible or not. |
| Type? | improvement |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | N/A |
| How to test? | N/A |

### CODE REVIEW

File: classes/AddressChecksumCore.php [hunk 1]
```diff
@@ -23,10 +23,22 @@
...
 *  International Registered Trademark & Property of PrestaShop SA
 */
+
+/**
+ * Class AddressChecksumCore
+ */
 class AddressChecksumCore implements ChecksumInterface
 {
+    /** @var string $separator Separator */
     private $separator = '_';
```
Comment (mickaelandrieu) [inline classes/AddressChecksumCore.php]: lets do better: would you mind to change it to a constant ?

```diff
- private $separator = '_';
+ const SEPARATOR = '_';
```

File: classes/AddressChecksumCore.php [hunk 2]
```diff
@@ -23,10 +23,22 @@
...
 
+    /**
+     * Generate a checksum
+     *
+     * @param Address $address
+     *
+     * @return string SHA1 checksum for the Address
+     */
     public function generateChecksum($address)
     {
```
Comment (mickaelandrieu) [inline classes/AddressChecksumCore.php]: By the way, this need a (little) update of this function too :)

Comment (firstred) [inline classes/AddressChecksumCore.php]: As in `SHA512`? ^^

Comment (mickaelandrieu) [inline classes/AddressChecksumCore.php]: Nope xD,

```diff
- $this->separator
+ self::SEPARATOR
```

everywhere :)

Comment (firstred) [inline classes/AddressChecksumCore.php]: Done :)

Comment (mickaelandrieu) [inline classes/AddressChecksumCore.php]: Merged :)

**Pull request merged**